### PR TITLE
Optimizations + code changes

### PR DESCRIPTION
**Pull request by Adrijaned**

### CODE REVIEW

File: src/main/java/org/terasology/Lakes/Lake.java [hunk 1]
```diff
@@ -115,16 +115,16 @@ public Vector3i getOrigin(){
         return origin;
     }
 
-    public int getWaterHeight() { return WaterHeight; }
+    /**
+     * Returns the y position of the surface od the lake.
```
Comment (skaldarnar) [inline src/main/java/org/terasology/Lakes/Lake.java]:
```suggestion
     * Returns the y position of the lake surface.
```

File: src/main/java/org/terasology/Lakes/LakeDepthFacet.java [hunk 2]
```diff
@@ -20,6 +20,9 @@
 import org.terasology.world.generation.Border3D;
 import org.terasology.world.generation.facets.SurfaceHeightFacet;
 
+/**
+ * Marks how deep a lake would be were there some in a particular world xz coordinates.
```
Comment (skaldarnar) [inline src/main/java/org/terasology/Lakes/LakeDepthFacet.java]: Does this facet provide the depth in absolute values or as difference to the "real" surface height?

If I read `LakeDepthProvider` correctly this facet returns a positive floating point value denoting the depth of the lake at a given position:

```suggestion
 * Marks how deep a lake would be at given world coordinates (x, z).
 *
 * The lake depth is given as positive floating point value which should be interpreted as depth from actual surface height.
```

**Review by skaldarnar — APPROVED** [review]
Thanks for the changes, code looks a bit cleaner and we can always make use of better documentation! I've slightly changed some of the doc comments and will merge it now 😃

**Review by skaldarnar — CHANGES_REQUESTED** [review]
@Adrijaned actually, I cannot apply the suggestions myself due to missing permissions (🤔 probably because that would mean to push to a branch on your fork...). 
Please be so kind to apply the suggestions (can do that via GitHub web interface) and ping me again 🙃

**Comment by Adrijaned** [comment]
@skaldarnar

**Review by skaldarnar — APPROVED** [review]
Thanks a lot!

**Pull request merged**